### PR TITLE
WEA-7 Implement easy permissions library

### PR DESCRIPTION
**Pull request by ayush-shta**

### CODE REVIEW

**Comment by subash-poudel** [comment]
This branch should have description: Should be merged only after merging PR WEA-6. Has a lot of repeated code with WEA-6. Didn't check.

File: app/src/main/java/com/ayush/weatherapp/home/HomeActivity.java [hunk 1]
```diff
@@ -155,4 +163,53 @@ private void setForecastDetails(DailyForecast.DailyData todaysForecast) {
...
+      presenter.fetchWeatherDetails();
+    } else {
+      EasyPermissions.requestPermissions(this, getString(R.string.request_permission_location),
+          RC_LOCATION_PERM, Manifest.permission.ACCESS_FINE_LOCATION);
+    }
+  }
+
+  @Override
+  public void onPermissionsGranted(int requestCode, @NonNull List<String> perms) {
+    Timber.e("onPermissionsGranted:" + requestCode + ":" + perms.size());
```
Comment (subash-poudel) [inline app/src/main/java/com/ayush/weatherapp/home/HomeActivity.java]: y

File: app/src/main/java/com/ayush/weatherapp/home/HomeContract.java [hunk 2]
```diff
@@ -1,21 +1,25 @@
...
 
+    void setLocality(String locality);
   }
 
   interface Presenter extends BaseContract.Presenter {
     View getView();
 
     void fetchWeatherDetails();
+
+    void fetchLocality(String requestString);
```
Comment (subash-poudel) [inline app/src/main/java/com/ayush/weatherapp/home/HomeContract.java]: parameter ambiguous

File: app/src/main/java/com/ayush/weatherapp/home/HomePresenterImpl.java [hunk 3]
```diff
@@ -47,17 +55,49 @@ public void onResponse(@NonNull Call<Forecast> call, @NonNull Response<Forecast>
...
+        getView().setCurrentForecast(currentForecast);
 
         getView().setDailyForeCast(dailyForecastList);
 
         getView().hideProgressDialog();
       }
 
       @Override public void onFailure(@NonNull Call<Forecast> call, @NonNull Throwable t) {
-        Timber.e("Request Failed");
+        Timber.e("Request fetch forecast failed");
```
Comment (subash-poudel) [inline app/src/main/java/com/ayush/weatherapp/home/HomePresenterImpl.java]: tim

File: app/src/main/java/com/ayush/weatherapp/home/HomePresenterImpl.java [hunk 4]
```diff
@@ -47,17 +55,49 @@ public void onResponse(@NonNull Call<Forecast> call, @NonNull Response<Forecast>
...
+    reverseGeoLocationCall.enqueue(new Callback<ReverseGeoLocation>() {
+      String localityAddress;
+
+      @Override
+      public void onResponse(Call<ReverseGeoLocation> call, Response<ReverseGeoLocation> response) {
+        ReverseGeoLocation reverseGeoLocation = response.body();
+        List<Address> addressList = reverseGeoLocation.getAddresses();
+        List<AddressComponents> addressComponentsList = addressList.get(0).getAddressComponents();
+
+        //Timber.e(reverseGeoLocation.getStatus());
```
Comment (subash-poudel) [inline app/src/main/java/com/ayush/weatherapp/home/HomePresenterImpl.java]: remove

File: app/src/main/java/com/ayush/weatherapp/home/HomeActivity.java (hunk 1 above)
Comment (ayush-shta) [reply app/src/main/java/com/ayush/weatherapp/home/HomeActivity.java]: :ok:

File: app/src/main/java/com/ayush/weatherapp/home/HomeContract.java (hunk 2 above)
Comment (ayush-shta) [reply app/src/main/java/com/ayush/weatherapp/home/HomeContract.java]: :ok:

File: app/src/main/java/com/ayush/weatherapp/home/HomePresenterImpl.java (hunk 3 above)
Comment (ayush-shta) [reply app/src/main/java/com/ayush/weatherapp/home/HomePresenterImpl.java]: :ok:

**Pull request merged**